### ISSUE TITLE
Questions about training procedure

### ISSUE THREAD

**Issue by wusuoweima** [canqin001/PointDAN#8]
I am a beginner in domain adaption and think there are some differences in the training procedure between your paper and code.
1. 
`             # Classification loss

            loss_s1 = criterion(pred_s1, label)

            loss_s2 = criterion(pred_s2, label)

            # Adversarial loss

            loss_adv = - 1 * discrepancy(pred_t1, pred_t2) 

            loss_s = loss_s1  +  loss_s2

            loss = args.weight * loss_s + loss_adv

            loss.backward()

            optimizer_g.step()

            optimizer_c.step()

            optimizer_g.zero_grad()

            optimizer_c.zero_grad()`
Are the above code corresponds to the below objective functions? 
![image](https://user-images.githubusercontent.com/30040019/75038663-c6f4b980-54f1-11ea-8f7d-65eaf12540d8.png)
1）Train G E F1 F2 by minimizing classification loss L_cls . 
2）Train F1 F2 by maximizing the discrepancy loss L_dis. 
3)Trian G E by  grad_reverse(`if adaptation == True:   x = grad_reverse(x, constant)` to minimizing the discrepancy loss L_dis. 

2.  
`            # Local Alignment

            feat_node_s = model(data, node_adaptation_s=True)

            feat_node_t = model(data_t, node_adaptation_t=True)

            sigma_list = [0.01, 0.1, 1, 10, 100]

            loss_node_adv = 1 * mmd.mix_rbf_mmd2(feat_node_s, feat_node_t, sigma_list)

            loss = loss_node_adv

            loss.backward()

            optimizer_dis.step()

            optimizer_dis.zero_grad()
`
Are the above code corresponds to the below objective functions? 
![image](https://user-images.githubusercontent.com/30040019/75038193-da535500-54f0-11ea-8cc1-05246ebe9bcb.png)
Train G E　G　R　by minimizing the　MMD　loss L_mmd.
3.Then, where are the below objective functions ?
![image](https://user-images.githubusercontent.com/30040019/75040468-a9c1ea00-54f5-11ea-8cfd-1a28254bdb1a.png)
i.e. train W R by minimizing the discrepancy loss, classification loss.
@canqin001 @Hxyou

**Comment by canqin001** [comment]
Hi. Thanks for your question.
In Part 1 (Classification Loss), it includes the min_{W,R} L_{cls}+\lambda*L_{dis}. In this part, the gradients of L{cls} and L{dis} would flow through the W and R by optimizer_g.step() without updating the offsets of SA Node.

**Comment by wusuoweima** [comment]
Thank you for your reply.  So is my understanding of Part1 and Part 2 correct?

**Comment by canqin001** [comment]
Excluding the W and R, it's right.

**Comment by wusuoweima** [comment]
Thanks
